x/mint/keeper/keeper.go

```diff
@@ -71,7 +62,7 @@ func NewKeeper(
 // queries. The method returns an error if current height in ctx is greater than the v7 upgrade height.
 func (k Keeper) SetInitialSupplyOffsetDuringMigration(ctx sdk.Context) error {
 	if !k.accountKeeper.HasAccount(ctx, k.accountKeeper.GetModuleAddress(types.DeveloperVestingModuleAcctName)) {
-		return errDevVestingModuleAccountNotCreated
+		return types.ErrDevVestingModuleAccountNotCreated
```

Comment:
I thought about wrapping this with sdkerrors like; 
```
sdkerrors.Wrapf(types.ErrDevVestingModuleAccountNotCreated, "%s module account already exists", types.DeveloperVestingModuleAcctName")
```

But then i'd have to change the way we're catching errors in keeper_test? Any suggestions 

cc @p0mvn

Reply:
Is refactoring to something like the following possible?
https://github.com/osmosis-labs/osmosis/blob/04424733e90c7b6f66264a9db6d71b48cb62ddba/x/mint/keeper/keeper_test.go#L636

Where the expected error is an sdk wrapper?

Reply:
1. I haven't seen sdkerrors used in any test at all, so wouldn't it be better to maintain consistency? 
2. Also, comparing sdkerrors in test seems very confusing, for instance this output is shown unequal 
```
 Not equal: 
 expected: vesting module account already exist: module account already exists [/Users/sishirg27/Desktop/osmosis/x/mint/keeper/keeper.go:86]
 actual  : vesting module account already exist: module account already exists [/Users/sishirg27/Desktop/osmosis/x/mint/keeper/keeper_test.go:238]
```

Reply:
1. I haven't really seen sdkerrors being used in test elsewhere, would i be better to maintain consistency? 
2. Comparing sdkerrors is super weird, for example this is showing unequal 
```
 Not equal: 
expected: vesting module account doesnot exist: module account does not exist [/Users/sishirg27/Desktop/osmosis/x/mint/keeper/keeper.go:66]
actual  : vesting module account doesnot exist: module account does not exist [/Users/sishirg27/Desktop/osmosis/x/mint/keeper/keeper_test.go:273]
                                    
    Diff:
    --- Expected
    +++ Actual
    @@ -8,7 +8,8 @@
       }),
    -  stack: (*errors.stack)((len=6) {
    +  stack: (*errors.stack)((len=7) {
        (uintptr) 0x4625d7b,
    -   (uintptr) 0x5061f88,
    -   (uintptr) 0x552c858,
    -   (uintptr) 0x5068896,
    +   (uintptr) 0x552c436,
    +   (uintptr) 0x40d1865,
    +   (uintptr) 0x40d0ddf,
    +   (uintptr) 0x5069656,
        (uintptr) 0x411a322,
```

Reply:
1. I think that's because we haven't been asserting the kinds of errors in tests in general. It is preferable to expect a specific error in tests to ensure we cover the desired branches in the tests.
2. This might work: `suite.ErrorIs(actualError, tc.expectedError)`

Let me know what you think

Reply:
THAT WORKED!!